libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt

```diff
@@ -83,7 +84,10 @@ class MapboxTripSession(
         set(value) {
             field = value
             ioJobController.scope.launch {
-                navigator.setRoute(value)
+                val status = navigator.setRoute(value)
+                mainJobController.scope.launch {
+                    isOffRoute = status.routeState == RouteState.OFF_ROUTE
+                }
```

Comment:
Why do we need this 🤔? We're already checking it as part of NN polling 👀 https://github.com/mapbox/mapbox-navigation-android/blob/907899a90eb70921641a9b6d6efe2cdcfc79d77f/libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt#L115 I saw you mentioned this https://github.com/mapbox/mapbox-navigation-android/pull/3184#issuecomment-654216075 but not fully understanding.

Reply:
This code hides potential issue, I've wrote it here https://github.com/mapbox/mapbox-navigation-android/issues/3291

Reply:
Hides it or solves it? 🤔

Reply:
hides. I'll remove current changes and investigate it separately, because issue is out of current PR